pkg/types/types.go

```diff
@@ -65,6 +65,7 @@ type Receipt struct {
 	BlockHash         common.Hash     `json:"blockHash,omitempty"`
 	BlockNumber       *big.Int        `json:"blockNumber,omitempty"`
 	TransactionIndex  uint            `json:"transactionIndex"`
+	EffectiveGasPrice *big.Int        `json:"effectiveGasPrice"`
```

Comment:
Consider doing an analysis on different EVM based chains to see if the field is always present, otherwise, it might be a good idea to use `omitempty`.